### PR TITLE
Debug Adapter Tracker Implementation

### PR DESCRIPTION
**Pull request by xnkevinnguyen**
# Description:

Implement the debug adapter tracker to log the state of the debugger and capture the messages.

The UI will react to the messages emitted by debug adapter tracker

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Limitations:

Please describe limitations of this PR

# Testing:

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] My code has been formatted with `npm run format` and passes the checks in `npm run check`
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/debugger/debugAdapter.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    }
+    onWillReceiveMessage(message: any): void {
+        if (message.command) {
+            // Only send pertinent debug messages
+            switch (message.command) {
+                case DEBUG_COMMANDS.CONTINUE:
+                    this.messagingService.sendStartMessage();
+                    break;
+                case DEBUG_COMMANDS.STACK_TRACE:
+                    this.messagingService.sendStopMessage();
```
Comment (isadorasophia) [inline src/debugger/debugAdapter.ts]: How is this different from `onWillStopSession()`? I think we actually want something that passes the idea that are on a broken state (e.g. `sendBreakMessage`?)

If we use "Stop" it gets the idea that we lose all our previous state.

**Review by isadorasophia — APPROVED** [review]
Can't really say much about the React bit, but LGTM 🍰

File: src/debugger/debugAdapter.ts (hunk 1 above)
Comment (xnkevinnguyen) [reply src/debugger/debugAdapter.ts]: makes sense! I will change this. The stack trace is send from the api before entering pause or breakpoint

**Review by isadorasophia — APPROVED** [review]

File: src/view/App.tsx [hunk 2]
```diff
@@ -39,20 +46,32 @@ class App extends React.Component<{}, IState> {
...
+                if (message.active_device !== this.state.currentDevice) {
+                    this.setState({ currentDevice: message.active_device });
+                }
+                break;
+            case VSCODE_MESSAGES_TO_WEBVIEW.RUN_DEVICE:
+                this.setState({ viewState: VIEW_STATE.RUNNING });
+                break;
+            case VSCODE_MESSAGES_TO_WEBVIEW.PAUSE_DEVICE:
+                this.setState({ viewState: VIEW_STATE.PAUSE });
+
```
Comment (vandyliu) [inline src/view/App.tsx]: _nit_ remove this empty line

File: src/view/components/microbit/MicrobitImage.tsx [hunk 3]
```diff
@@ -51,14 +69,30 @@ const setupButton = (
     buttonElement.onmouseleave = e => {
         eventTriggers.onMouseLeave(e, key);
     };
+    console.log("buttons should be enabled");
```
Comment (vandyliu) [inline src/view/components/microbit/MicrobitImage.tsx]: _nit_ remove this console.log

**Review by vandyliu — APPROVED** [review]
LGTM

**Pull request merged**